### PR TITLE
Improve focus handling in spec-editor directive

### PR DESCRIPTION
**Pull request by algairim**
Signed-off-by: Mykola Mandra <mykola.mandra@cloudsoftcorp.com>

### CODE REVIEW

File: ui-modules/blueprint-composer/app/components/spec-editor/spec-editor.directive.js [hunk 1]
```diff
@@ -181,6 +187,10 @@ export function specEditorDirective($rootScope, $templateCache, $injector, $sani
             }
         };
 
+        scope.state = sessionStorage && sessionStorage.getItem(scope.model._id)
+            ? JSON.parse(sessionStorage.getItem(scope.model._id))
+            : defaultState;
```
Comment (algairim) [inline ui-modules/blueprint-composer/app/components/spec-editor/spec-editor.directive.js:192]: Moved up to enable external configuration, e.g. branding.

File: ui-modules/blueprint-composer/app/components/spec-editor/spec-editor.directive.js [hunk 2]
```diff
@@ -406,11 +414,18 @@ export function specEditorDirective($rootScope, $templateCache, $injector, $sani
             }
             return $filter('specEditorConfig')(allConfig, scope.state.config.filter.values).indexOf(config) === -1;
         };
+        specEditor.recordFocus = (subsection, name) => {
+            scope.state.focus.subsection = subsection;
+            scope.state.focus.name = name;
+        };
```
Comment (algairim) [inline ui-modules/blueprint-composer/app/components/spec-editor/spec-editor.directive.js:420]: A single subsection must be in focus only. Previously, only the last subsection was focused on refresh (when DSL menu closes) if multiple expected to be in focus, like property or configuration at the same time.

File: ui-modules/blueprint-composer/app/components/spec-editor/spec-editor.directive.js [hunk 3]
```diff
@@ -106,7 +110,7 @@ export function specEditorDirective($rootScope, $templateCache, $injector, $sani
 
     function controller($scope, $element) {
         (composerOverrides.configureSpecEditorController || function () {
-        })(this, $scope, $element);
+        })(this, $scope, $element, blueprintService);
```
Comment (algairim) [inline ui-modules/blueprint-composer/app/components/spec-editor/spec-editor.directive.js:113]: Blueprint service is required for branding configuration.

**Comment by algairim** [comment]
Screenshot 2021-06-03 at 09 16 29
Screenshot 2021-06-03 at 09 18 15
Screenshot 2021-06-03 at 09 18 41
Screenshot 2021-06-03 at 09 19 05
Screenshot 2021-06-03 at 09 19 18

**Review by iuliana — APPROVED** [review]
LGTM, but I would feel more comfortable if somebody with more UI expertise would take a look too.

**Review by duncangrant — APPROVED** [review]
LGTM - thanks @algairim

**Comment by ahgittin** [comment]
nice cleanup

**Pull request merged**